CMakeLists.txt

```diff
@@ -50,92 +73,29 @@ find_package(Boost COMPONENTS
   filesystem
   REQUIRED)
 find_package (Threads)
-
-set(SMFC_SOURCES
-  src/smfc/codegen.cc
-  src/smfc/cpp_generator.cc
-  src/smfc/go_generator.cc
-  src/smfc/main.cc)
-
-set(SMFC_LIBS
-  Flatbuffers::flatbuffers
-  Boost::system
-  Boost::filesystem
-  gflags
-  glog::glog
-  ${CMAKE_THREAD_LIBS_INIT}
-  )
-
-add_executable(smfc ${SMFC_SOURCES})
-target_link_libraries(smfc PRIVATE ${SMFC_LIBS})
-target_include_directories(smfc PRIVATE src/smfc)
-
-
-find_package(Zstd REQUIRED)
 find_package(Hdrhistogram REQUIRED)
 find_package(xxHash REQUIRED)
-find_package(Seastar 1.0 REQUIRED)
-
-smfc_gen(
-  CPP
-  TARGET_NAME raw_rpc_gen
-  OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/src/include/smf
-  SOURCES ${PROJECT_SOURCE_DIR}/src/core/rpc.fbs)
-
+find_package(Zstd REQUIRED)
 
-file(GLOB smf_includes "${CMAKE_CURRENT_SOURCE_DIR}/src/include/*.h")
-file(GLOB smf_srcs "${CMAKE_CURRENT_SOURCE_DIR}/src/core/*.cc")
-add_library(smf STATIC
-  ${raw_rpc_gen}
-  ${smf_srcs}
-  ${smf_includes}
-  )
-set_target_properties(smf PROPERTIES VERSION ${PROJECT_VERSION})
-target_include_directories(smf PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/include>
-    $<INSTALL_INTERFACE:include>  # <prefix>/include
-)
-if(CMAKE_BUILD_TYPE MATCHES Debug)
-  target_compile_options(smf PRIVATE
-    -fdiagnostics-color=auto
-    -Wall
-    -Werror
-    -Wextra
-    -Wformat
-    -Wmissing-braces
-    -Wparentheses
-    -Wpointer-arith
-    -Wformat-security
-    -Wunused
-    -Wno-unused-parameter
-    -Wcast-align
-    -Wno-missing-field-initializers
-    -Wdelete-non-virtual-dtor
-    -Wno-ignored-qualifiers)
+if(NOT SMF_ENABLE_TESTS)
+  set(SMF_ENABLE_INTEGRATION_TESTS  OFF)
+  set(SMF_ENABLE_UNIT_TESTS  OFF)
+  set(SMF_ENABLE_BENCHMARK_TESTS  OFF)
 endif()
 
-target_compile_definitions(smf PUBLIC -DXXH_PRIVATE_API)
-target_link_libraries(smf
-  PUBLIC
-  Seastar::seastar
-  Flatbuffers::flatbuffers
-  xxHash::xxhash
-  Zstd::zstd
-  Hdrhistogram::hdr_histogram)
+# put all the libs and binaries in one place
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
```

Comment:
what's the difference between archive and lib? also, do you need to mention a lib64?

Reply:
i believe that archive files would be static libraries (https://cmake.org/cmake/help/v3.0/prop_tgt/ARCHIVE_OUTPUT_DIRECTORY.html). as for lib64 -- in my experience this is easier to handle at packaging time (e.g. lib64 is centos/rhel thing)